### PR TITLE
Allow using custom processor in async mode

### PR DESCRIPTION
**Pull request by linyao-reevo**
Allow use to specify a custom processor in async mode:
```ruby
          Pheromone.setup do |config|
            config.background_processor.name = :custom
            config.background_processor.klass = 'CustomJob'
            config.background_processor.custom_processor = -> (klass, msg) do
              klass.perform(msg)
            end
          end
```
@ankitagupta12

### CODE REVIEW

File: spec/pheromone/messaging/message_dispatcher_spec.rb [hunk 1]
```diff
@@ -93,6 +93,65 @@ def perform(topic:, message:, metadata: {}, options: {})
...
+          expect do
+            described_class.new(
+              message_parameters: message_parameters,
+              dispatch_method: :async
+            ).dispatch
+          end.to raise_error
+        end
+      end
+    end
+
```
Comment (z0lope0z) [inline spec/pheromone/messaging/message_dispatcher_spec.rb]: extra space

File: lib/pheromone/messaging/message_dispatcher.rb [hunk 2]
```diff
@@ -38,6 +40,10 @@ def send_message_asynchronously
           Resque.enqueue(background_processor_klass, message_body)
         elsif background_processor.name == :sidekiq
           background_processor_klass.perform_async(message_body)
+        elsif background_processor.name == :custom
+          background_processor.custom_processor.call(background_processor_klass, message_body)
```
Comment (ankitagupta12) [inline lib/pheromone/messaging/message_dispatcher.rb]: does this need to be `constantize`d. `klass` in the setting above is a string so should we be consistent?

**Comment by ankitagupta12** [comment]
Could you also add this feature to the `Readme.md`?

File: lib/pheromone/messaging/message_dispatcher.rb (hunk 2 above)
Comment (ankitagupta12) [inline lib/pheromone/messaging/message_dispatcher.rb]: does this need to be `constantize`d. `klass` in the setting above is a string so should we be consistent?

File: lib/pheromone/config.rb [hunk 3]
```diff
@@ -12,6 +12,8 @@ class Config
       setting :name
       # specify the background job handling message send to kafka
       setting :klass
+      # specify custom background job
+      setting :custom_processor
```
Comment (ankitagupta12) [inline lib/pheromone/config.rb]: let's add the type for this setting so it's clear it's a proc

**Review by ankitagupta12 — APPROVED** [review]

**Pull request merged**